### ISSUE TITLE
[BUG] DATABASE URL inconsistencies

### ISSUE THREAD

**Issue by kevinpapst** [tobybatch/kimai2#545]
With the recent change, that `?charset=utf8&serverVersion=5.7` is required in the `DATABASE_URL` we now have some issues.

- The single env vars do not supply those right now, we need to add `DB_CHARSET` and `DB_VERSION` in multiple files, e.g. here:
  - [ ]  https://github.com/tobybatch/kimai2/blob/main/assets/service.sh#L15
  - [ ] https://github.com/tobybatch/kimai2/blob/main/assets/startup.sh#L25

- ServerVersion and Charset are missing:
  - [ ] https://github.com/tobybatch/kimai2/blob/main/compose/traefik/docker-compose.yml#L49
  - [ ] https://github.com/tobybatch/kimai2/blob/main/Dockerfile#L244
  - [ ] https://github.com/tobybatch/kimai2/blob/main/Makefile#L56
  - [ ] https://github.com/tobybatch/kimai2/blob/main/.github/workflows/test-images.yml#L36

- Different database version used in regular vs traefik images (mysql:5 vs mysql:latest)
  -  [ ] DISCUSS: Upgrade normal images to use mysql:8 ?

I can work on PRs, especially the first one, but I need some input for the ones regarding the tests and test-image @tobybatch .

**Comment by hexxone** [comment]
Hey there, I have another question.

In the README, you are providing the "serverVersion" of "5.7". 
However, you are also using "mysql:latest" TAG.
And if I am correct, the "latest" mysql Docker Version is already at something like > `8.0.X`.

Should these versions correlate? 
And if so, should the "DATABASE_URL" Version be incremented, or the mysql Version be "statically" set instead?

Also, it would be nice to have these "breaking changes" in the Release section of kimai, because I had to manually search and figure out this is now a thing.... ^^

Thanks in advance!

**Comment by kevinpapst** [comment]
mysql:latest is only used in the traefik image, which is not maintained by tobybatch.
So 5.7 should be correct for the normal docker images (I don't even know what Traefik is).
But I agree that should be at least discussed, so I added it to the todo list.

Downgrading a database isn't really possible, so going back to 5 is obviously no option.
Using "latest" is probably also not the best idea for prod images.
So I would suggest that we find out the exact version from mysql:latest and use that in all docker images as static version eg mysql:8.

> and figure out this is now a thing

This was always a thing. But nobody read the .env template properly in the last years. 

> it would be nice to have these "breaking changes" in the Release section of kima

Totally agree. Mainly my fault. The examples in the documentation were all super old and missing the `charset=utf8mb5&serverVersion=5.7` examples. So I updated all documentations (not docker, only Kimai itself!) and the [UPGRADING guide](https://github.com/kimai/kimai/blob/5c139a7b75c7cf8496ed2095c4489970e0aa045d/UPGRADING.md) will contain the breaking change soon.

Thanks for sharing your findings @hexxone 👍
